pkgs/top-level/python-packages.nix

```diff
@@ -11419,18 +11419,29 @@ self: super: with self; {
     cudaSupport = pkgs.config.cudaSupport or false;
     inherit (pkgs.darwin.apple_sdk.frameworks) CoreServices;
     inherit (pkgs.darwin) libobjc;
+    inherit (pkgs.llvmPackages_rocm) openmp;
   };
 
   torch-bin = callPackage ../development/python-modules/torch/bin.nix { };
 
   torchWithCuda = self.torch.override {
+    magma = pkgs.magma-cuda;
     cudaSupport = true;
   };
 
   torchWithoutCuda = self.torch.override {
     cudaSupport = false;
   };
 
+  torchWithRocm = self.torch.override {
+    magma = pkgs.magma-hip;
+    rocmSupport = true;
+  };
+
+  torchWithoutRocm = self.torch.override {
+    rocmSupport = false;
+  };
```

Comment:
@mweinelt should we add more of them or not? I am undecided.

Reply:
No strong opinion. It would be nice if it would be cached, but we depend on unfree bits here, no?

Reply:
The rocm version should be completely free.

Reply:
As far as torch (with rocm) being cached at all, I'm unsure that's possible since we apparently need to propagate rocfft along with other rocm libs.
We can't cache rocfft because of hydra limits, each .so is ~700MB, and unless we up the limit at least for rocfft or build it twice and link, it just isn't going to work.

Reply:
why does it need to be built twice? Can the package not just be split and then symlinked together again?

Reply:
Or does split output not increase the limit on hydra cache?

Reply:
IIRC, the hydra limit includes the sum of all outputs.
So, I can't just put say, 2 of the .so files into ``${rocfft.lib}`` and link them into ``${rocfft}``.
We would have to build twice and remove two of the .so files from one and link them in the other to work around this, if we could not just either increase the hydra limit or make a special exception for rocfft.

Reply:
It would be nice to be able to build once and have each .so file be something we could "fetch" to avoid this like with miopen's kdb files, but I don't know of a feasible way to do this in nixpkgs.
Would it be possible to build in a ``let`` prefix and then ``runCommand`` copy .so files to their own outputs, and then link those to a total output?

Reply:
Hydra limits outputs to 2GB right now, does it not fit into that?

Reply:
Hydra limits are currently 3GB.
rocFFT is... 3.8GB.
Because this is a package for a distribution, it has to be general so I can't just limit what devices the shaders in the library files are generated for either.
I'm currently experimenting with having each .so have it's own derivation in the hopes we can work around this assuming we can't just up the limit or grant an exception.

Reply:
Unfortunately I don't think hydra has any support  for a `meta.hydraMaxOutputSize` or anything

Reply:
Marking this as resolved because the rocFFT caching issue has seemingly been solved and there are no unfree bits for the rocm variant.